### DISCUSSION TITLE
GHOSTS Command Server Specs?

### DISCUSSION THREAD

**Discussion by crispy6** [cmu-sei/GHOSTS#173] [Q&A] [answered]
Hi everyone! I'm looking for the suggested specs (CPUs, RAM, storage, etc.) for the VM that I'll be running the GHOSTS command server on. Any recommendations? Thank you!

**Comment by sei-dupdyke** [comment, marked as answer]
This is a good question, and I need to add this to the documentation! Thank you!

We have been running this configuration in our greyspace, over a perhaps a dozen ranges (in varying degrees of use) for few years now, using standard docker setup:

```
Guest OS:                          CentOS 7 (64-bit)
CPUs:                              2 CPU(s)
Reservation:                       0 MHz
Limit:                             Unlimited
Hardware virtualization:           Disabled 
Memory:                            16 GB
Hard disk 1:                       64 GB
Video card:                        8 MB 
```

We have used on Ubuntu as well, and no issues there, with the move to Centos8, I'd probably just recommend Ubuntu these days.

We roll the database to a new one with logical breaks in exercises just to segment off data, and you might do this by exercise, month, year, or whatever suits you best. Our last rollover  had 100s of millions of rows of history with no known performance problems. We typically set clients to send activity to the API every 15 minutes.

HTH

**Reply by crispy6** [reply]
Thank you for the thorough response!
